assembly/compliance-tool/src/test/java/org/eclipse/sw360/antenna/frontend/compliancetool/sw360/updater/SW360UpdaterTest.java

```diff
@@ -61,38 +58,21 @@ public void testClearingReportGeneratorMustNotBeNull() {
 
     @Test
     public void testUpdateWithFailure() throws IOException {
-        SW360Release testRelease = SW360TestUtils.mkSW360Release("test");
-
         SW360UpdaterImpl updaterImpl = mock(SW360UpdaterImpl.class);
-        when(updaterImpl.artifactToReleaseInSW360(any(), any()))
-                .thenReturn(testRelease);
-        when(updaterImpl.artifactToReleaseWithUploads(any(), any(), any()))
-                .thenReturn(new AttachmentUploadResult<>(testRelease));
 
         String propertiesFilePath = Objects.requireNonNull(this.getClass().getClassLoader().getResource("compliancetool-updater.properties")).getPath();
         final Map<String, String> properties = ComplianceFeatureUtils.mapPropertiesFile(new File(propertiesFilePath));
         initConfiguration(properties);
 
-        SW360ReleaseClientAdapter releaseClientAdapter = mock(SW360ReleaseClientAdapter.class);
-
-        SW360Connection connection = mock(SW360Connection.class);
-        when(connection.getReleaseAdapter())
-                .thenReturn(releaseClientAdapter);
-        when(configurationMock.getConnection()).thenReturn(connection);
-
-        // the creation of the clearing document will failed one time and be successful one time
         final ClearingReportGenerator clearingReportGenerator = mock(ClearingReportGenerator.class);
         when(clearingReportGenerator.createClearingDocument(any(), any()))
-                .thenThrow(new SW360ClientException("Clearing doc generation error"))
-                .thenReturn(folder.newFile("clearing_document.test").toPath());
+                .thenThrow(new SW360ClientException("Clearing doc generation error"));
         SW360Updater updater = new SW360Updater(updaterImpl, configurationMock, clearingReportGenerator);
 
-        // and an exception is nevertheless thrown by SW360ReleaseClientAdapterAsyncImpl
         Assertions.assertThatThrownBy(updater::execute)
                 .isInstanceOf(SW360ClientException.class);
 
-        // but we expect that attachment upload worked one time exactly
-        verify(releaseClientAdapter, times(1)).uploadAttachments(any());
+        verify(updaterImpl, never()).artifactToReleaseWithUploads(any(), any(), anyMap());
```

Comment:
One major goal of this test was to verify whether exceptions thrown during the processing of an artifact are caught and do not cause the process to abort. Therefore, the clearing report generator was configured to throw an error once and then succeed. In the second iteration, a successful upload should be performed.

With the reworked test, this is no longer the case. Was there a reason to strip down the test in this way?

Reply:
From the updater, I removed the, in my opinion, redundant second upload of attachments. But with that the method `uploadAttachments` is never called by the `SW360Updater` anymore. Consequently, I removed this from the test and changed it to verify that the method `artifactToReleaseWithUploads` is never called. Furthermore, I removed other mocking calls that are not called anymore.

Reply:
After a discussion with @nnobelis I finally understood the intention of the test. Please have a look at the changes.